tools/check_examples/common.py

```diff
@@ -0,0 +1,115 @@
+"""Common helper functions."""
+
+import sys
+import subprocess
+from pathlib import Path
+
+PASS = "✅"
+FAIL = "❌"
+WARN = "⚠️ "
+INDENT = 4
+IGNORE_FILE = "ignore.txt"
+
+ignore_list: list[Path] = []
+
+
+def iprint(message: str, indent: int) -> None:
+    """Print with indent level."""
+    lines = message.splitlines(True)
+    indented_lines = [" " * INDENT * indent + line for line in lines if line.strip()]
+    block = "".join(indented_lines)
```

Comment:
Each of the line has `\n` at the end because `splitlines()` is called with `True`.  The purpose of the list comprehension line is to remove blank lines to make the output more compact.  `block` is a bunch of lines each with `\n` at the end.